portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -101,4 +113,69 @@ public SchedulingAlgorithmType getSchedulingAlgorithmType() {
 
     return scheduledTimeRangeOptional;
   }
+
+  /**
+   * Keeps splitting up the task to schedule the currently avaible free time ranges until all of the
+   * task is scheduled across the different time ranges. When this method is called, the task may or
+   * may not be complete scheduled.
+   *
+   * @return a boolean value representing whether or not the task is complete scheduled.
+   */
+  private boolean splitUpTaskToSchedule(
+      Task task, TimeRangeGroup availableTimesGroup, List<ScheduledTask> scheduledTasks) {
+    List<TimeRange> currentAvailableTimes = constructAvailableTimeRanges(availableTimesGroup);
+    Duration taskDuration = task.getDuration();
+    int taskSegmentCount = 1;
+
+    for (TimeRange currentFreeTimeRange : currentAvailableTimes) {
+      if (taskDuration.getSeconds() > 0) {
+        Instant scheduledTime = currentFreeTimeRange.start();
+
+        // Reconstruct the task segment's name.
+        String taskName = task.getName() + " (Part " + taskSegmentCount + ")";
+        String taskDescription = "";
+        if (task.getDescription().isPresent()) {
+          taskDescription = task.getDescription().get();
+        }
+        TaskPriority taskPriority = task.getPriority();
+
+        // If task's current duration is longer than the free time's,
+        // this means the entirety of the free time range is scheduled to this task.
+        if (taskDuration.compareTo(currentFreeTimeRange.duration()) > 0) {
+          // Constructs a new taks object with the current free time range's duration.
+          Task taskWithActualScheduledDuration =
+              new Task(taskName, taskDescription, currentFreeTimeRange.duration(), taskPriority);
+          ScheduledTask scheduledTask =
+              new ScheduledTask(taskWithActualScheduledDuration, scheduledTime);
+          scheduledTasks.add(scheduledTask);
+
+          availableTimesGroup.deleteTimeRange(currentFreeTimeRange);
+          taskDuration = taskDuration.minus(currentFreeTimeRange.duration());
+        } else {
+          // Otherwise, only part of the free time range is needed to schedule this task.
+          Task taskWithActualScheduledDuration =
```

Comment:
I think we might be able to use a helper method that can be used in both if and else cases.

ex:
private void scheduleTask(Task task, Instant scheduledTime) {
  ScheduledTask scheduledTask = ...;
  scheduledTasks.add(scheduledTask);
  Instant scheduledTaskEndTime = ...;
  TimeRange scheduledTaskTimeRange = ...;
  availableTimesGroup.deleteTimeRange(...);
  return ScheduledTask;
}

Also, I think we should update ScheduledTask so it exposes a method that returns the scheduled task time range and scheduled task end time. Then we can simplify the helper method further.

Then in the if/else branches we can do,

ScheduledTask scheduledTask = scheduleTask(task, ...);
taskDuration = taskDuration.minus(task.getDuration());

Reply:
Updated.